app/models/miq_cockpit_ws_worker/authenticator.rb

```diff
@@ -0,0 +1,53 @@
+class MiqCockpitWsWorker::Authenticator
+  def authenticate_for_host(token, host)
+    user_obj = user_for_token(token)
+    return {} unless user_obj
+
+    # is it a container deployment
+    creds = find_container_node_creds(user_obj, host)
+
+    # Is it a VM
+    creds = find_vm_creds(user_obj, host) if creds.nil?
+
+    {
+      :valid  => true,
+      :known  => !creds.nil?,
+      :key    => !creds.nil? && creds.auth_key,
+      :userid => !creds.nil? && creds.userid,
+    }
+  end
+
+  private
+
+  # TODO: Do we need more user based permissions checks?
+  def find_container_node_creds(user_obj, host_or_ip)
+    raise "Looking up container nodes requires a valid user" unless user_obj
+    cond = "container_deployment_nodes.name = ? OR container_deployment_nodes.address = ?"
+    deployment = ContainerDeployment.joins(:container_deployment_nodes)
+                                    .find_by(cond, host_or_ip, host_or_ip)
```

Comment:
@NickLaMuro thanks...

> @petervo @jrafanie At a minimum, we should probably be doing this as a .where, because you can still do prepared statements from that:

```
ContainerDeployment.joins(:container_deployment_nodes)
                   .where(cond, host_or_ip, host_or_ip).first
```

Really?  I didn't know that.  I thought `find_by` is very similar to `where(...).first`.